### ISSUE TITLE
Chunked prefill crashes with mlx-lm >= 0.31.0: zip unpacking expects 6 values, gets 7

### ISSUE THREAD

**Issue by omprashantjain** [waybarrios/vllm-mlx#155]
## Bug Description

When using `vllm-mlx` with `mlx-lm >= 0.31.0` and `--continuous-batching` enabled, chunked prefill crashes with a `ValueError` because the `zip(*batch_prompts)` unpacking in `scheduler.py` expects exactly 6 tuple elements, but mlx-lm 0.31.x now returns 7.

## Root Cause

mlx-lm PR [ml-explore/mlx-lm#911](https://github.com/ml-explore/mlx-lm/pull/911) ("Better caching in the server") added a `prompt_checkpoints` field to the prompt tuples returned by `BatchGenerator.unprocessed_prompts`. This means each entry in `batch_prompts` now contains 7 elements instead of 6.

The unpacking at [scheduler.py L378-L385](https://github.com/waybarrios/vllm-mlx/blob/main/vllm_mlx/scheduler.py#L378) hardcodes exactly 6 target variables:

```python
(
    uids,
    inputs_raw,
    max_tokens_list,
    caches,
    samplers,
    logits_processors,
) = zip(*batch_prompts)
```

## Steps to Reproduce

1. Install `vllm-mlx==0.2.6` with `mlx-lm>=0.31.0`
2. Start with `--continuous-batching` enabled
3. Send any inference request
4. Observe crash/ValueError on the `zip(*batch_prompts)` line

## Environment

- **vllm-mlx**: 0.2.6
- **mlx-lm**: 0.31.1
- **mlx**: 0.31.1
- **Hardware**: Apple Silicon (Mac Studio Ultra, M2 Ultra 192GB)
- **Model**: Qwen3.5-35B-A3B-8bit (also reproduced with Qwen3-14B)

## Proposed Fix

Add a `*_extra` catch-all to the unpacking so it's forward-compatible with additional fields from mlx-lm:

```python
(
    uids,
    inputs_raw,
    max_tokens_list,
    caches,
    samplers,
    logits_processors,
    *_extra,
) = zip(*batch_prompts)
```

This is a minimal, non-breaking change — the extra values are captured but ignored, maintaining backward compatibility with older mlx-lm versions that return 6 elements.

PR incoming with the fix.

**Comment by Thump604** [comment]
This is fixed by PR #221 (krystophny, "scheduler: preserve prompt checkpoints in chunked prefill resume path"). The PR addresses the recent mlx-lm contract change where prompt tuples now carry a seventh `prompt_checkpoints` field that the chunked-prefill monkeypatch needs to retain across partial resume. PR is currently CONFLICTING (needs a rebase on current main) but the fix itself targets the same root cause as your `zip(*batch_prompts)` unpack failure.

**Comment by Thump604** [comment]
Fixed on current main across multiple commits:

1. **Tuple unpacking** (`cace3f3`, `6b28ab1`): `_chunked_next` at `scheduler.py:480-488` now unpacks 7 elements including `prompt_checkpoints`
2. **API compatibility** (`980d092`): `_install_chunked_prefill` checks `hasattr(bg, "_process_prompts")` before patching, gracefully disabling chunked prefill on incompatible mlx-lm versions
3. **Return type handling** (`980d092`): `BatchGenerator.next()` result is checked — `isinstance(result, tuple)` handles both the new `(prompt_responses, generation_responses)` format and the old flat list

Verified on commit `2cce3da`.
